.github/workflows/main.yml

```diff
@@ -11,16 +11,69 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        libvirt: ['3.1.0', '5.1.0']
+        include:
+          - libvirt: '3.1.0'
+            os: 'ubuntu-18.04'
+          - libvirt: '5.1.0'
+            os: 'ubuntu-18.04'
+
+    runs-on: ${{ matrix.os }}
+    env:
+      CCACHE_TEMPDIR: /tmp/.ccache-temp
+      EXT: xz
+      LIBVIRT: ${{ matrix.libvirt }}
 
     steps:
       - uses: actions/checkout@v2 # respository path: $GITHUB_WORKSPACE
       - uses: actions/setup-go@v2
         with:
           go-version: 1.16.2
 
-      - name: Install dependencies
-        run: go get -u golang.org/x/lint/golint
+      - name: Install golint and goyacc
+        run: |
+          go get -u golang.org/x/lint/golint
+          go get golang.org/x/tools/cmd/goyacc
+
+      - name: Enable apt source package downloads and update index
+        run: |
+          sudo sed -i '/deb-src/s/^# //' /etc/apt/sources.list
+          sudo apt update
+
+      - name: Build libvirt from source and install it
+        run: |
+          sudo apt-get -qqy build-dep libvirt
+          sudo apt-get -qqy install curl ccache qemu-system-x86
+          sudo mkdir -p /usr/src && sudo chown $(id -u) /usr/src
+          curl -O -s https://libvirt.org/sources/libvirt-${LIBVIRT}.tar.${EXT}
+          tar -C /usr/src -xf libvirt-${LIBVIRT}.tar.${EXT}
+          pushd /usr/src/libvirt-${LIBVIRT}
+          ccache --show-stats
+          env PATH=/usr/lib/ccache:$PATH \
+             ./configure --prefix=/usr --localstatedir=/var --sysconfdir=/etc \
+                         --without-polkit \
+                         --without-esx --without-vbox --without-xen --without-libxl --without-lxc \
+                         --with-qemu
+          make
+          sudo make install
+          ccache --show-stats
+          popd
+          sudo libvirtd -d -l -f $GITHUB_WORKSPACE/.github/testdata/libvirtd.conf
+          sudo virtlogd -d || true
+
+      - name: Setup test artifacts
+        env:
+          TESTDATA: ${{ github.workspace }}/.github/testdata
```

Comment:
Its not common to put test artifacts in `.github/`, and testdata is independent of github or travis (because it used to be in .travis), maybe it should just be at the top-level, where its easily viewable, maintainalbe, etc? Which would also remove the need for `TESTDATA` as an env var.

Reply:
So looking through all that's in there (as I didn't change any of it, just moved it from a travis directory to a github directory), the libvirtd.conf is specific to the CI test setup and run, so I think that file at least should remain alongside the workflows that use them, and not the integrations tests.

The other things I think are tied to the integration tests themselves, and I'm okay moving them closer to those; however, I don't like just dumping everything in the top level directory and cluttering it.  I'd rather put a 'testdata' directory in the top level and not the files themselves.

Reply:
> put a 'testdata' directory in the top level

Yes, that's what I'd suggest.